Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -191,3 +191,18 @@ \section{Concurrency}
 These seperate threads will get a specific task assigned, which has his local data to compute on.
 Deadlocks will be prevented because the spawned threads will be joined before acquiring locks in synchronized statements.
 This will happen when we persist the algorithm output in the database.
+
+\section{Scalability}
+Concurrent clients are one thing, but we also have to scale the algorithms for growth.
+Initially, we planned on a nearest neighbours algorithm to match songs to rooms.
+Later on, we found out that while this works well for small datasets, finding the nearest neighbours in a system of a few million songs can prove to be a lot harder.
```

Comment:
I would not amend "(while this works well for small datasets)" because its already made clear by " a few million songs". The quantity of the requests with a large user base, and the realtime performance requirements could be added .